Adding the Generational Data Interview Series

clare-stanton opened the pull request:
This PR is to add the Generational Data Interview Series to the LIL website. 

This interview series was created as a discrete editorial project, and thus has its own styling and information system that is different from other parts of the LIL website. There is a project landing page, and then 14 interviews that are connected via a navigation system as well as editorial notes that bring readers from one interview to another. 

To accommodate that while still having the series sit within the LIL website, I created a new layout and css file for Jekyll to use when generating pages. There is now a new folder in /app labeled _interviews, that holds 14 markdown files for each of the 14 interviews conducted. 

There is also a new folder, generational-data-interviews, that holds assets like headshots, and graphics used throughout the series; custom fonts; an index.html for the landing page of the series; a custom .js file for the series; and a readme that explains this structure and has instructions for adding new interviews. 

Major functionality includes:
- A generated list of interviewee names on the landing page 
- A generated scroll of quote cards also on the landing page 
- A generated list of interviewee names on the individual interviews 
- A system for navigating between the individual interviews 
- An "editorial note" that opens when a reader hovers over highlighted text 

**Note:** _This work was assisted by Cursor, the AI coding tool. While efforts were made to keep the .js and .css in particular as compact and efficient as possible, reviewers should be aware that the primary focus of this work initially was to get to the needed functionality._ 

Known issues and remaining work: 
- There is an issue with the dropdown menu of interviews in the mobile version of the individual interview pages. It keeps closing on its own. 
- There is an instance of an interview with two interviewees. Their combined names are long and sometimes interrupt the design, like on the landing page when their names are listed in the interviews section. 
- Accessibility: adding ARIA landmarks, improving keyboard nav 
- I think the swup integrations are causing a bit of a lag in the loading of navigation arrows and interview lists 

Note about the Github Actions commits: this was an attempt to share the code for QA that we abandoned and opted to share locally instead!

clare-stanton committed:
Add interview series structure and initial interviews.

- Add generational data interviews to Our Work page
- Add folder for Generational Data Interviews assets, landing page, and js
- Create new layout template for individual interviews
- Create custom css for interviews and landing page
- Add interview pages to _config.yml
- Create folder for interview markdown files

clare-stanton committed:
Adding
- more custom css
- javascript for navigation and pull quotes
- custom font
- graphics for landing page
- rest of the interview copy

clare-stanton committed:
CSS cleanup and HTML improvements for interview series

clare-stanton committed:
Documentation rewrite and interview formatting improvements

- Rewrite of README.md to include file locations and data structure
- Updated Frank Cifaldi interview to use HTML speaker tags for consistent styling

clare-stanton committed:
Add GitHub Actions workflow for Pages deployment

- Configure automatic deployment from interview-series branch
- Build Jekyll site from app/ directory
- Deploy to GitHub Pages for QA testing

clare-stanton committed:
Fix GitHub Actions build issues

- Add missing jekyll gem to Gemfile
- Add CSS build step for PostCSS/Tailwind processing
- Ensure proper build order: npm install -> CSS build -> Jekyll build

clare-stanton committed:
Fix GitHub Actions bundle install issues

- Disable bundler cache to avoid conflicts
- Add explicit bundler installation step
- Create GitHub Pages compatible config without jekyll-postcss-v2
- Use simplified Gemfile for deployment
- Separate build process for CSS and Jekyll

clare-stanton committed:
Improve GitHub Actions build process

- Switch from PostCSS to direct Tailwind CSS build
- Add interview series paths to Tailwind config
- Add debugging output to identify build failures
- Add error handling for CSS build step
- Improve build process reliability

clare-stanton committed:
Revert GitHub Pages deployment changes

- Remove GitHub Actions workflow files
- Remove GitHub Pages specific config files
- Revert Gemfile to original state
- Revert Tailwind config to original state
- Return to pre-deployment state

clare-stanton committed:
Editorial feature updates and CSS Reorg

Added
- navigation at bottom of interviews
- rewritten editor's note feature
- pull quote styling
- content of pull quotes to individual interviews
- link styling
- URLs to interview bios

Reorganized
- CSS file
- Readme for interview series

clare-stanton committed:
Remove large build file

clare-stanton committed:
Remove debugging code from main.js

Remove debug code

clare-stanton committed:
Editorial updates and some more CSS work

Editorial
- added editors notes to each interview
- rearragned pull quotes to sit properly in the interviews
- added most recent introduction copy

Styling/Layout
- Changed the way the introduction columns populate
- Fixed font coloring for interviews
- Worked on mobile breakpoints
- worked on responsive sizing for pull quotes

ascourtas commented on .github/workflows/test-build-deploy.yml:
Is this meant to be deleted? this config controls github workflows and CI

clare-stanton replied:
Nope! Do not know what happened there.

clare-stanton replied:
Aha - it got deleted when I was trying to remove my own GitHub action jazz.

ascourtas commented on app/_interviews/amelia-acker.md:
```diff
@@ -0,0 +1,63 @@
+
+<p>There's a lot of historians of science who say that the best bureaucracies are just entities that you can kick and are not going to go away. They're going to stick around for a while, whether we like them or not, whether they're ugly or aesthetically pleasing. So oftentimes they take up space, they're not flexible and adaptive in some ways.</p>
+
+<p><a href="/generational-data-interviews/rebecca-cremona/" class="editor-note-link" target="_blank">My understanding is also that bureaucracies and institutions assert legitimacy in an interesting way. Digital tools and collections are pretty slippery, putting your finger on what legitimacy and authenticity is in most of the spaces that we're talking about is actually hard for most of us.</a><sp ...
```
I can't tell if this block is doing what it's supposed to -- it looks like in the middle of Acker's answer is either a link to a statement from Becky, or at least a shift to the interviewer without the proper formatting? unclear

clare-stanton replied:
Yeah, this is an "Editor's Note", which is a lil text box that shows up in the margins when a user hovers over some highlighted text. They have commentary and point to other interviews in the series. 

<img width="1137" height="378" alt="Screenshot 2025-10-09 at 4 52 06 PM" src="https://github.com/user-attachments/assets/d9493653-8378-4753-9915-9dff66b6477b" />

ascourtas replied:
Nice! from an HTML formatting perspective it's definitely a little funky, if you're concerned at all about our future selves looking at this I'd recommend adding a comment.

ascourtas commented on app/_interviews/frank-cifaldi.md:
```diff
@@ -0,0 +1,63 @@
+<div class="speaker-label">Frank Cifaldi</div>
+<p>I would have the best copyright lawyers in the country figuring out how we can actually make this work.</p>
+
+Second, from a tech perspective, I would invest in the ability for researchers to tinker with, build, and play historical video games, to gain a better understanding of what makes them click, what makes them unique.
```
this file doesn't use p tags like the others do -- I'd recommend making it consistent

clare-stanton replied:
Good catch! I tried a couple of different ways to get these markdown files to work correctly, and missed some things

ascourtas commented on app/_interviews/rebecca-frank.md:
```diff
@@ -0,0 +1,53 @@
+
+<p>I think if you just give the money to the people who are doing the work and that probably gets you 90% of the way to a solution. It's not super exciting technically or in any other way, but right now the biggest problem is nobody knows if they'll be able to do what they're doing three years from now because that's when their grant runs out. They can't maintain staffing with the technical exper ...
+
+<p><a href="/generational-data-interviews/amelia-acker/" class="editor-note-link" target="_blank">I also think maybe the answer is to take that money and move it away from the institutions, away from government, away from organizations and send it to communities that have a very real stake in preserving their own types of information.</a><span class="editor-note">In contrast, Amelia Acker made a  ...
```
similar comment as before -- just want to check and make sure this is doing what you want it to be doing! 🌈

ascourtas commented on app/_layouts/interview.html:
```diff
@@ -0,0 +1,74 @@
+    <section class="interview-page__profile" style="padding-top: 120px; background: #EFEEEB;">
+        <div class="container">
+            <div class="interview-page__profile-content" style="padding: 0 20px;">
+                <div class="interview-page__profile-image">
```
not a necessary change, but stylistically double underscore is more meant for naming element types (assuming this is using a [BEM](https://getbem.com/naming/) approach?)

ascourtas commented on app/_layouts/interview.html:
```diff
@@ -0,0 +1,74 @@
+                        <div class="interview-page__text-nav-next">
+                            <span class="interview-page__text-nav-name" id="text-next-name"></span>
+                            <a href="#" class="interview-page__nav-arrow interview-page__nav-arrow--next" id="text-next-arrow">
+                                <span>→</span>
```
does this display well on mobile?

clare-stanton replied:
If this is the part of the page that I think it is, great question! I added some custom behavior for mobile that doesn't use the names: 

<img width="405" height="732" alt="Screenshot 2025-10-09 at 4 49 06 PM" src="https://github.com/user-attachments/assets/594c79ba-7770-4255-af4e-0d62926fbe6c" />

ascourtas commented on app/generational-data-interviews/README.md:
```diff
@@ -0,0 +1,240 @@
+| **Interview Pages** | `app/_interviews/*.md` | Individual interview content using Jekyll markdown |
+| **Layout Template** | `app/_layouts/interview.html` | Jekyll layout for interview pages |
+| **JavaScript Engine** | `generational-data-interviews/main.js` | Dynamic content loading, navigation, and interactions |
+| **Styling** | `app/assets/css/interview-series.css` | Complete visual design system |
```
> Complete visual design system

These descriptions are a little over the top and definitely read as AI-generated -- only important if we expect people outside of LIL to look at the repo? but if not, I think it's fine

ascourtas replied:
I hold this sentiment for most of the README here

ascourtas commented on app/generational-data-interviews/main.js:
```diff
@@ -0,0 +1,1209 @@
+// Generational Data Interviews - Main JavaScript File
```
heads up I am not gonna look at this file, it's so meaty 😅 I am very suspicious of 1200 lines just to render the interview tiles, pull quotes, etc and would strongly recommend adding a comment or md file that links to the Chat conversation record if possible, or at least mentions this was AI-generated and briefly describes the process. I'm thinking of our future selves that might need to either debug this if it breaks, or, who might try to copy this code for something else and can't get it to work. 

Unfortunately this file is where a failure is most likely, but it's also where validating the code is the hardest. JavaScript can be funky even at the best of times.

clare-stanton replied:
Yeah I like the note generally that the individual files should be flagged as AI assisted - one of the pitfalls of coding this way was that I know enough to experiment and get to a clean place, but it was hard to always make sure that the experimental stuff was getting cleaned back 100%.

ascourtas replied:
yeah that sounds tricky!

ascourtas commented on app/generational-data-interviews/main.js:
```diff
@@ -0,0 +1,1209 @@
+// Generational Data Interviews - Main JavaScript File
+// Handles all interview series functionality
```
I'd also recommend adding a comment about what functionality is actually covered here

ascourtas reviewed (commented):
Hey @clare-stanton great work! Very neat that you could vibe code this. I would strongly recommend adding comments to all AI-gen files that they are in fact AI-gen -- I think this will be important for our future selves, either if this breaks or if people copy code for another part of the website. I have more details about this in the `main.js` comment.

*Addendum -- Musings on Main*
I am definitely finding myself longing for days of DRY, concise code in this age of AI-gen code. I know you asked me to primarily look at the HTML files but major bugs and design decisions won't be made there -- they'll be in the JavaScript. However a 1200 line JS file is kind of impenetrable, which creates a black box problem of "well I don't know what's in it, because AI wrote it, but I also can't validate it, because it's so much content". I don't know if there's really a way around this without just forgoing AI-coding? It does concern me for the future of reliable software, and for software bloat in general. But the good news is this PR doesn't have to address our societal infrastructure problems :) Just sharing thoughts

clare-stanton committed:
Cleaning up stray HTML in markdown, added AI notes

- Markdown files contained HTML, which is removed except for editors notes. Switched up styling to point at markdown instead.
- Added into to major files about use of Cursor
- Also added back in GitHub tests that were accidentally deleted while trying to clean up GitHub pages attempt :)
